scipy/sparse/tests/test_base.py

```diff
@@ -4300,7 +4300,10 @@ def spmatrix(self, D, **kwargs):
                                            zero_pos[0][k],
                                            zero_pos[1][k])
 
-        arg1 = self._arg1_for_noncanonical(M)
+        if sorted_indices:
+            arg1 = self._arg1_for_noncanonical(M, sorted_indices)
+        else:
+            arg1 = self._arg1_for_noncanonical(M)
```

Comment:
I'm not sure why this condition exists. Seems like we could just call
```
self._arg1_for_noncanonical(M, sorted_indices=sorted_indices)
```
in either case.